The-SMIRNOFF-force-field-format.md

```diff
@@ -1,129 +1,345 @@
+Some top-level tags provide attributes that modify the functional form used to be consistent with packages such as AMBER or CHARMM.
+
+
+## Charge models
 
-## Functional forms, etc.
+SMIRNOFF supports several ways for specifying electrostatic models.
+Currently, only classical fixed point charge models are supported, but extensions to the specification will support point multipoles, point polarizable dipoles, Drude oscillators, charge equilibration methods, and so on.
 
-**Functional form**: The SMIRNOFF format specifies parameters; once specified, these are processed by the SMIRNOFF `ForceField` class and used to assign parameters to OpenMM Forces.
-This means that specific forces are generally implemented as discussed in the [OpenMM Documentation](http://docs.openmm.org/7.0.0/userguide/theory.html), see especially [Section 19 on Standard Forces](http://docs.openmm.org/7.0.0/userguide/theory.html#standard-forces) for functional forms. In some cases, typically for consistency with the AMBER force field philosophy motivating some of the authors, we do some manipulation of parameters from these files as discussed below in "Parameter sections".
+### Library charges via `<LibraryCharges>`
 
-**Charges**: In keeping with the AMBER force field philosophy, especially as implemented in small molecule force fields such as [GAFF](http://ambermd.org/antechamber/gaff.html), [GAFF2](https://mulan.swmed.edu/group/gaff.php), and [parm@Frosst](http://www.ccl.net/cca/data/parm_at_Frosst/), we at least initially treat partial charges as something to be obtained separately from the rest of the force field (bonds, angles, impropers, torsions [BAIT] and vdW terms), typically via QM calculations or a method such as Bayly's [AM1-BCC](https://dx.doi.org/10.1002/jcc.10128) approach, thus, for system setup we provide the option of specifying a charging method, though charges are not normally specified in the FFXML itself.
- For other force fields with "library"-style charges, we could introduce a new section providing specific charges via SMIRKS pattern, though this is not yet implemented. Bond charge corrections, however, are supported as we discuss below.
+A mechanism is provided for specifying library charges that can be applied to molecules or residues that match provided templates.
+Library charges are applied first, and atoms for which library charges are applied will be excluded from alternative charging schemes listed below.
+
+For example, to assign partial charges
+```XML
+<LibraryCharges charge_unit="elementary_charge">
+   <!-- match a non-terminal alanine residue -->
+   <LibraryCharge name="ALA" smirks="[$([NX3H:1](C)(C)[H:2])][CX4H:3]([CH3X4:4]([H:5][H:6][H:7]))[CX3:8](=[OX1:9])([N])" charge1="-0.4157" charge2="0.2719" charge3="0.0337" charge4="-0.1825" charge5="0.0603" charge6="0.0603" charge7="0.0603" charge8="0.5973" charge9="-0.5679">
```

Comment:
This SMIRKS has a number of problems. 
* The `$` will confuse a human reader
* `[H:5][H:6][H:7]` means that hydrogen6 is bound to hydrogen5 and hydrogen7 is bound to hydrogen6, not what you were trying to get across. 

I suggest: `smirks = "[NX3:1]([#1:2])([#6])[#6X4H1:3]([#6:4]([#1:5])([#1:6])[#1:7])[#6:8](=[#8:9])[#7]"`

However, the `H` decorator has an implied 1, which doesn't work if the ALA is at the N-terminus. Similarly the last `N` makes this not match an ALA at the C-terminus.
If you want the smirks to match the termiuses, then I would use:
`smirks = "[NX3:1]([#1:2])[#6X4H1:3]([#6:4]([#1:5])([#1:6])[#1:7])[#6:8](=[#8:9])"`

Reply:
Thanks!

I think I was missing the alpha-carbon proton earlier, so how about this modification of your suggestion?
```XML
   <!-- match a non-terminal alanine residue -->
   <LibraryCharge name="ALA" smirks="[NX3:1]([#1:2])([#6])[#6H1:3]([#1:4])([#6:5]([#1:6])([#1:7])[#1:8])[#6:9](=[#8:10])[#7] charge1="-0.4157" charge2="0.2719" charge3="0.0337" charge4="0.0823" charge5="-0.1825" charge6="0.0603" charge7="0.0603" charge8="0.0603" charge9="0.5973" charge10="-0.5679">
```

Reply:
That should be fine